File: c/euler3.c
/*
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143 ?
*/
#include <stdio.h>

int main(){
  long n;
  // n = 600851475143;
  // n = 13195;
  n = 51;
  int i = 3;
  int j;
  int lpf;
  int primeflag;
  while(1){
    if (n % i == 0) { // if it's a factor
      // check to see if the number is prime
      primeflag = 1;
      for(j = 2; j < i/2; j ++){
        if (i % j == 0){
          primeflag = 0;
          break;
        }
      }
      if (primeflag){
        lpf = i;
        n /= i;
      }
    }
    if (i > n){
      break;
    }
    i += 2;
  }
  printf("%d is the largest prime factor\n", lpf);
}

File: python/euler3.py
# euler3.py

# Solves problem 3 on Project Euler

# The prime factors of 13195 are 5, 7, 13, and 29.
# What is the largest prime factor of the number 600851475143

"""SOLVED"""

# the idea for finding the largest prime factor of n is to successively divide n
# by integers, and then check to see if the integers in question are prime.

def main():
	# n = 27
	# n = 13195
	n = 600851475143
	largestprimefactor = 2
	i = 3
	while True:
		if n % i == 0:
			for j in range(2, i/2):
				if i % j == 0:
					break
			else:
				largestprimefactor = i
				n = n/i
		if i > n:
			break
		i += 2

	print "%d is the largest prime factor" %(largestprimefactor)

main()
